### PR TITLE
Stopping the animation when unmounting

### PR DESCRIPTION
**Pull request by moQuez**
### Overview

This PR is intended to stop the shimmer animation when the component gets unmounted.

#### Why?

TL;DR;... It never notifies the native UI thread that an animation ended, breaking apps that expect this to happen.

I stumbled upon this issue when using https://github.com/wix/detox on a project and testing a loading part of the app using shimmer loading. The problem is that the shimmer placeholder actually never stopped the animation. When just looking at the app, it seems to work ok, but when inspecting the native UI thread and listening to lifecycle changes, it starts to misbehave.

#### Solution

This PR is pretty simple. It accomplishes the intended behavior just by changing two thins
1. implement a componenWillUnmount and stop the shimmer animation
2. when looping, pay attention to the Animated callback args to see if the animation was stopped on purpose and prevent starting it again

### CODE REVIEW

**Review by AndreiCalazans — APPROVED** [review]
Awesome, this is also a performance improvement for apps that use this library intensively. Great job man!
